lib/internal/Magento/Framework/Registry.php

```diff
@@ -8,7 +8,11 @@
 /**
  * Registry model. Used to manage values in registry
  *
+ * Registry usage as a shared service introduces temporal, hard to detect coupling into system.
+ * It's usage should be avoid. Use service classes or data providers instead.
```

Comment:
Some good example of service class and data provider wouldn't hurt here I believe.

Reply:
Just any implementation of service class or data provider which looks good in your eyes.

Ok, probably wider explanation makes sense in release notes where this will be actually deprecated and not in code comments.